src/fit.jl

```diff
@@ -120,7 +120,7 @@ function train!(
     end
 
     # save the model in serialised form, in case we should be deepcopied or serialised elsewhere
-    estimator.model = LGBM_BoosterSaveModelToString(estimator.booster, 0, 0)
+    estimator.model = LGBM_BoosterSaveModelToString(estimator.booster, 0, 0, 0)
```

Comment:
need to re-think usage of the new parameter. as per API update it's the `feature_importance_type` parameter indicating the type of feature importance, where `0` is for `C_API_FEATURE_IMPORTANCE_SPLIT`and `1` for `C_API_FEATURE_IMPORTANCE_GAIN`

Reply:
I added a test to cover this case and it seems there is no difference if either of them is provided as a parameter. The test generates a model, calculates split and gain, then uses the `LGBM_BoosterSaveModelToString` to same model and then to re-loads it. At each stage the split and gain are checked vs the expected outputs.